tests/unit/sagemaker/cli/compatibility/v2/modifiers/test_py_version.py

```diff
@@ -0,0 +1,145 @@
+        "sagemaker.chainer.estimator.Chainer({})",
+        "PyTorch({})",
+        "sagemaker.pytorch.PyTorch({})",
+        "sagemaker.pytorch.estimator.PyTorch({})",
+        "SKLearn({})",
+        "sagemaker.sklearn.SKLearn({})",
+        "sagemaker.sklearn.estimator.SKLearn({})",
+    )
+
+
+@pytest.fixture
+def constructor_model_templates():
+    return (
+        "MXNetModel({})",
+        "sagemaker.mxnet.MXNetModel({})",
+        "sagemaker.mxnet.model.MXNetModel({})",
+        "ChainerModel({})",
+        "sagemaker.chainer.ChainerModel({})",
+        "sagemaker.chainer.model.ChainerModel({})",
+        "PyTorchModel({})",
+        "sagemaker.pytorch.PyTorchModel({})",
+        "sagemaker.pytorch.model.PyTorchModel({})",
+    )
+
+
+@pytest.fixture
+def constructor_templates(constructor_framework_templates, constructor_model_templates):
+    return tuple(list(constructor_framework_templates) + list(constructor_model_templates))
+
+
+@pytest.fixture
+def constructors_no_version(constructor_templates):
+    return (ctr.format("") for ctr in constructor_templates)
+
+
+@pytest.fixture
+def constructors_with_version(constructor_templates):
+    return (ctr.format("py_version='py3'") for ctr in constructor_templates)
+
+
+@pytest.fixture
+def constructors_with_image_name(constructor_framework_templates):
+    return (ctr.format("image_name='my:image'") for ctr in constructor_framework_templates)
+
+
+@pytest.fixture
+def constructors_with_image(constructor_model_templates):
+    return (ctr.format("image='my:image'") for ctr in constructor_model_templates)
+
+
+@pytest.fixture
+def constructors_version_not_needed():
+    return (
+        "TensorFlowModel()",
+        "sagemaker.tensorflow.TensorFlowModel()",
+        "sagemaker.tensorflow.model.TensorFlowModel()",
+        "SKLearnModel()",
+        "sagemaker.sklearn.SKLearnModel()",
+        "sagemaker.sklearn.model.SKLearnModel()",
+    )
+
+
+def _test_modified(constructors, should_be):
+    modifier = py_version.PyVersionEnforcer()
+    for constructor in constructors:
+        node = ast_call(constructor)
+        if should_be:
+            assert modifier.node_should_be_modified(node)
+        else:
+            assert not modifier.node_should_be_modified(node)
+
+
+def test_node_should_be_modified_fw_constructor_no_version(constructors_no_version):
+    _test_modified(constructors_no_version, should_be=True)
+
+
+def test_node_should_be_modified_fw_constructor_with_version(constructors_with_version):
+    _test_modified(constructors_with_version, should_be=False)
+
+
+def test_node_should_be_modified_fw_constructor_with_image_name(constructors_with_image_name):
+    _test_modified(constructors_with_image_name, should_be=False)
+
+
+def test_node_should_be_modified_fw_constructor_with_image(constructors_with_image):
+    _test_modified(constructors_with_image, should_be=False)
+
+
+def test_node_should_be_modified_fw_constructor_not_needed(constructors_version_not_needed):
```

Comment:
it doesn't seem like any of these fixtures are used in multiple tests, so I'm not sure what the benefit is to defining fixtures for them rather than just creating the iterable in the test directly

Reply:
i was envisioning using them with `framework_version` tests in a separate pr.

Reply:
gotcha - that makes sense!